### PR TITLE
Migrate extension Config to `@ConfigMapping`

### PR DESCRIPTION
**Pull request by gastaldi**
/cc @geoand @gsmet @holly-cummins @radcortez

### CODE REVIEW

**Comment by gastaldi** [comment]
It fails with 
```
Caused by: java.lang.ClassNotFoundException: io.quarkus.workshop.superheroes.version.deployment.VersionConfig
	at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:641)
	at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:188)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:525)
	at io.quarkus.bootstrap.classloading.QuarkusClassLoader.loadClass(QuarkusClassLoader.java:533)
	at io.quarkus.bootstrap.classloading.QuarkusClassLoader.loadClass(QuarkusClassLoader.java:481)
	at io.quarkus.bootstrap.classloading.QuarkusClassLoader.loadClass(QuarkusClassLoader.java:533)
	at io.quarkus.bootstrap.classloading.QuarkusClassLoader.loadClass(QuarkusClassLoader.java:481)
	at io.quarkus.runtime.configuration.AbstractConfigBuilder.withMapping(AbstractConfigBuilder.java:64)
	... 21 more
```


Not sure what's causing this, investigating

**Comment by dmlloyd** [comment]
Run time config interfaces must be in the runtime artifact, not the deployment artifact; could it be that simple?

**Comment by radcortez** [comment]
That is the most likely issue :)

**Comment by gastaldi** [comment]
Weird how that always worked with @ConfigRoot, but that makes sense

File: quarkus-workshop-super-heroes/super-heroes/extension-version/deployment/src/main/java/io/quarkus/workshop/superheroes/version/deployment/ExtensionVersionProcessor.java [hunk 1]
```diff
@@ -19,7 +20,7 @@ FeatureBuildItem feature() {
     @BuildStep
     @Record(ExecutionTime.RUNTIME_INIT)
     void recordVersion(ApplicationInfoBuildItem app, VersionConfig versionConfig, VersionRecorder recorder) {
-        if (versionConfig.enabled) {
+        if (versionConfig.enabled()) {
```
Comment (geoand) [inline quarkus-workshop-super-heroes/super-heroes/extension-version/deployment/src/main/java/io/quarkus/workshop/superheroes/version/deployment/ExtensionVersionProcessor.java:23]: If the config is runtime, this shouldn't be done in the processor

Comment (gastaldi) [reply quarkus-workshop-super-heroes/super-heroes/extension-version/deployment/src/main/java/io/quarkus/workshop/superheroes/version/deployment/ExtensionVersionProcessor.java:23]: Just to confirm: should we pass `VersionConfig` to the `io.quarkus.workshop.superheroes.version.runtime.VersionRecorder#printVersion` and call the `versionConfig.enabled()` there?

**Review by geoand — APPROVED** [review]

**Comment by dmlloyd** [comment]
Previously it was not explicitly bound to a phase, so I guess it defaulted to "build time", which is probably why it worked before.

**Comment by geoand** [comment]
+1

**Comment by gastaldi** [comment]
Looks like it's ready to be merged now

**Pull request merged**